types/rdf-ext/lib/Dataset.d.ts

```diff
@@ -1,13 +1,13 @@
-import { Dataset } from 'rdf-js';
+import { BaseQuad, Dataset } from 'rdf-js';
 import { PropType } from './_PropType';
 import QuadExt = require('./Quad');
 
-interface DatasetExt extends Dataset {
+interface DatasetExt<Q extends BaseQuad = QuadExt> extends Dataset<Q> {
   readonly length: number;
   toJSON(): Array<ReturnType<PropType<QuadExt, 'toJSON'>>>;
```

Comment:
Because of this line, does this mean it should be `Q extends QuadExt`?

Reply:
I think it should be a closed type

```
interface DatasetExt extends Dataset<QuadExt> 
```

rdf-ext will always use its `Ext` types AFAICT

Reply:
Same for https://github.com/DefinitelyTyped/DefinitelyTyped/blob/1163ead296d84e7a3c80d71e7c81ecbd1a130e9a/types/rdf-ext/lib/DataFactory.d.ts#L42-L43 too?

Reply:
I should think that it does not make sense to make anything in rdf-ext generic. Those implementations as specific specialisations of the base interface